distributed/shuffle/_rechunk.py

```diff
@@ -658,20 +658,22 @@ def __init__(
 
     async def _receive(
         self,
-        data: list[tuple[NDIndex, list[tuple[NDIndex, tuple[NDIndex, np.ndarray]]]]],
+        input_partitions: list[NDIndex],
+        output_partitions: list[NDIndex],
+        data: list[tuple[NDIndex, np.ndarray]],
```

Comment:
We could probably zip this as well, but let's see if we gain anything from what we have done so far.

Reply:
https://github.com/dask/distributed/pull/8738/commits/48be30f169e58c7d20949d2ce0e6699305d3a85e